config/v1/types_cluster_version.go

```diff
@@ -406,8 +420,19 @@ type ComponentOverride struct {
 type URL string
 
 // Update represents an administrator update request.
+// +kubebuilder:validation:XValidation:rule="has(self.architecture) && has(self.version) ? (size(self.architecture) == 0 || size(self.version) == 0) : true",message="cannot set both Architecture and Version"
 // +k8s:deepcopy-gen=true
 type Update struct {
+	// architecture is an optional field that indicates the desired
+	// value of the cluster architecture. In this context cluster
+	// architecture means either a single architecture (homogeneous)
+	// or a multi architecture (heterogeneous). Architecture is
+	// ignored if image is specified. The empty value indicates that
+	// the field is not set.
```

Comment:
What happens if this is not set?  Does the CVO inspect the nodes in my cluster and make the best choice it can?